### ISSUE TITLE
Sitemap filename can't exceed 32 characters

### ISSUE THREAD

**Issue by erikhansen** [magento/magento2#13937]
Issue: Sitemap filenames can't exceed 32 characters.

### Preconditions
1. Install Magento Open Source 2.2.3

### Steps to reproduce
1. Go to **MARKETING > Site Map**
2. Click **Add Sitemap**
3. Enter a filename that exceeds 32 characters (for example `thisisareallylongsitemapnamethatexceeds32characters.xml`)
![image](https://user-images.githubusercontent.com/129031/36936698-3e4d82f2-1ece-11e8-9158-f8e6451569e2.png)
4. Click **Save & Generate**

### Expected result
1. The `thisisareallylongsitemapnamethatexceeds32characters.xml` file should get created in the `/pub/` directory.

### Actual result
1. The success message includes a truncated filename:
![magento admin-h1b3c](https://user-images.githubusercontent.com/129031/36936739-c04d5d4a-1ece-11e8-9c22-7108ba7515e4.jpg)
2. Sitemap file gets truncated:
![current ssh -peo5f](https://user-images.githubusercontent.com/129031/36936754-df4de746-1ece-11e8-9224-27b6aa8cbafd.jpg)

### Additional Details

The `sitemap_filename` field is a `VARCHAR(32)`, which explains why this issue is happening:

```
CREATE TABLE `sitemap` (
  `sitemap_id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Sitemap Id',
  `sitemap_type` varchar(32) DEFAULT NULL COMMENT 'Sitemap Type',
  `sitemap_filename` varchar(32) DEFAULT NULL COMMENT 'Sitemap Filename',
  `sitemap_path` varchar(255) DEFAULT NULL COMMENT 'Sitemap Path',
  `sitemap_time` timestamp NULL DEFAULT NULL COMMENT 'Sitemap Time',
  `store_id` smallint(5) unsigned NOT NULL DEFAULT '0' COMMENT 'Store id',
  PRIMARY KEY (`sitemap_id`),
  KEY `SITEMAP_STORE_ID` (`store_id`),
  CONSTRAINT `SITEMAP_STORE_ID_STORE_STORE_ID` FOREIGN KEY (`store_id`) REFERENCES `store` (`store_id`) ON DELETE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=7 DEFAULT CHARSET=utf8 COMMENT='XML Sitemap';
```

**Comment by erikhansen** [comment]
I just tested changing the VARCHAR length limit on the sitemap field. After running the query below, I was able to create a sitemap name that was up to 255 characters in length.

```
ALTER TABLE `sitemap` CHANGE `sitemap_filename` `sitemap_filename` VARCHAR(255)  CHARACTER SET utf8  COLLATE utf8_general_ci  NULL  DEFAULT NULL  COMMENT 'Sitemap Filename';
```

**Comment by magento-engcom-team** [comment]
@erikhansen, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by osrecio** [comment]
Hi @erikhansen ,

We can adjust the column length and add validation in the form. 

I think 255 chars should be enough to store a filename, following [Limits of filename](https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits)

What do you think?

**Comment by erikhansen** [comment]
@osrecio I agree that 255 characters should be sufficient.

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
Hi @erikhansen. Thank you for your report. 
The issue has been fixed in magento/magento2#20044 by @irajneeshgupta in 2.3-develop branch
Related commit(s):
 - [55cf2dcb970f3f069e3d59ad7986413f80bb6cdb](https://github.com/magento/magento2/commit/55cf2dcb970f3f069e3d59ad7986413f80bb6cdb)
 - [035e8d736961c1a074b8a732ec3bbd993670aacc](https://github.com/magento/magento2/commit/035e8d736961c1a074b8a732ec3bbd993670aacc)
 - [417ffa2cd273e32f9b0231e37cb49ec04af5a0ad](https://github.com/magento/magento2/commit/417ffa2cd273e32f9b0231e37cb49ec04af5a0ad)
 - [e9f4981181d13e33c286e293ab6829f9db504780](https://github.com/magento/magento2/commit/e9f4981181d13e33c286e293ab6829f9db504780)
 - [94c176fa5cd1047187b1478945cbec663d283d62](https://github.com/magento/magento2/commit/94c176fa5cd1047187b1478945cbec663d283d62)

The fix will be available with the upcoming 2.3.2 release.

**Comment by Ctucker9233** [comment]
Does a new backport need to be created?
